src/wallet/wallet.cpp

```diff
@@ -2833,6 +2839,9 @@ DBErrors CWallet::LoadWallet(bool& fFirstRunRet)
     if (nLoadWalletRet != DB_LOAD_OK)
         return nLoadWalletRet;
     fFirstRunRet = !vchDefaultKey.IsValid();
+    if (IsWalletFlagSet(WALLET_FLAG_DISABLE_HOT_KEYS)) {
```

Comment:
Agree, and combine the two conditionals to make it clearer that that's what is happening.

```
fFirstRunRet = !vchDefaultKey.IsValid() && !IsWalletFlagSet(WALLET_FLAG_DISABLE_HOT_KEYS);
```